flink-state-backends/flink-statebackend-changelog/src/main/java/org/apache/flink/state/changelog/ChangelogKeyedStateBackend.java

```diff
@@ -506,7 +517,9 @@ public void notifyCheckpointAborted(long checkpointId) throws Exception {
                         stateChangelogWriter,
                         meta,
                         stateDesc.getTtlConfig(),
-                        stateDesc.getDefaultValue());
+                        stateDesc.getDefaultValue(),
+                        ++lastCreatedStateId);
```

Comment:
I wonder how the `lastCreatedStateId` work with restored state? Will it increase to the restored state id?

Reply:
There is no connection between "old" state id (before recovery) and the "new" one (after recovery).
Recovery doesn't depend on the new state id as the mapping (from id to name) is written in snapshot (though recovery might increment `lastCreatedStateId`).
New snapshot after recovery will not depend on the old id because the mapping will be written at least once after recovery.